cpp/src/fil/fil.cu

```diff
@@ -581,7 +581,7 @@ void check_params(const forest_params_t* params, bool dense)
   ASSERT(params->blocks_per_sm >= 0, "blocks_per_sm must be nonnegative");
   ASSERT(params->n_items >= 0, "n_items must be non-negative");
   ASSERT(params->threads_per_tree > 0, "threads_per_tree must be positive");
-  ASSERT(thrust::detail::is_power_of_2(params->threads_per_tree),
+  ASSERT(params->threads_per_tree & (params->threads_per_tree - 1) == 0,
```

Comment:
This failed tests because the operator precedence for `==` is higher than bitwise and `&`. Oops! I need to wrap this in another set of parentheses.
```suggestion
  ASSERT((params->threads_per_tree & (params->threads_per_tree - 1)) == 0,
```